base/deprecated.jl

```diff
@@ -1722,6 +1722,9 @@ end
 @deprecate mapreduce(f, op, v0, itr) mapreduce(f, op, itr; init=v0)
 @deprecate mapfoldl(f, op, v0, itr) mapfoldl(f, op, itr; init=v0)
 @deprecate mapfoldr(f, op, v0, itr) mapfoldr(f, op, itr; init=v0)
+# also deprecate the old deprecations
+@deprecate mapreduce(f, op, v0, A; dims=dims) mapreduce(f, op, A; init = v0, dims=dims)
```

Comment:
I think this definition overwrites the one 4 lines above, as there is no dispatch on keywords (I also made the same mistake a couple of times). While compiling julia, there is a message that `mapreduce` and `reduce` are overwriten. The problem is that now I get an error message in a package using `mapreduce` that `dims` is not defined.

Reply:
Here too:
```julia
julia> reduce(+, 0, [1,2,3])
ERROR: UndefVarError: dims not defined
Stacktrace:
 [1] reduce(::Function, ::Int64, ::Array{Int64,1}) at ./deprecated.jl:54
 [2] top-level scope at none:0
```

Reply:
Hmm, is there no way to do this then? I'll have a look, sorry!